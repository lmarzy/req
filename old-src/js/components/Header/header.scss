// ========================================================================
// JS/COMPONENTS/HEADER
// ========================================================================

@import '~sassHelpers';

@keyframes logo {
  0% {
    transform: translateX(100px);
  }

  100% {
    opacity: 1;
    transform: translateX(0);
  }
}

@keyframes tagline {
  0% {
    transform: translateX(-100px);
  }

  100% {
    opacity: 1;
    transform: translateX(0);
  }
}

.header {
  display: flex;
  height: 100vh;
  padding-top: 20vh;
  padding-left: spacer(2);
  padding-right: spacer(2);
  justify-content: center;
  background-image: linear-gradient(to right top, $col-a, $col-b);
  clip-path: polygon(0 0, 100% 0, 100% 80%, 0 100%);
}

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.logo,
.tagline {
  opacity: 0;
}

.logo {
  display: inline-block;
  color: white;
  font-size: font-size(40);
  font-weight: 700;
  text-shadow: 1px 1px 3px rgba(black, 0.3);
  animation: logo 1000ms ease-in forwards;

  @include bp(10) {
    font-size: font-size(50);
  }

  @include bp(20) {
    font-size: font-size(60);
  }
}

.tagline {
  color: lighten($col-b, 25);
  font-size: font-size(20);
  text-shadow: 1px 1px 2px rgba(black, 0.4);
  animation: tagline 1000ms ease-in forwards;

  @include bp(10) {
    font-size: font-size(30);
  }
}