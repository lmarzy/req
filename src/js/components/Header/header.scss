// ========================================================================
// JS/COMPONENTS/PARTIALS/HEADER
// ========================================================================

@import '~sassHelpers';

@keyframes logo {
  0% {
    transform: translateX(100px);
  }

  100% {
    opacity: 1;
    transform: translateX(0);
  }
}

@keyframes tagline {
  0% {
    transform: translateX(-100px);
  }

  100% {
    opacity: 1;
    transform: translateX(0);
  }
}

.header {
  display: flex;
  height: 100vh;
  padding-top: 20vh;
  padding-left: spacer(2);
  padding-right: spacer(2);
  justify-content: center;
  background-image: linear-gradient(to right top, $col-a, $col-b);
  clip-path: polygon(0 0, 100% 0, 100% 65%, 0 100%);
}

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.logo,
.tagline {
  opacity: 0;
}

.logo {
  display: inline-block;
  color: lighten($col-b, 35);
  font-family: Monaco, Courier New, monospace;
  font-size: font-size(xl);
  font-weight: 700;
  text-shadow: 1px 1px 3px rgba(black, 0.3);
  animation: logo 1000ms ease-in forwards;

  @include bp(10) {
    margin-bottom: spacer(1);
    font-size: font-size(xxl);
  }
}

.tagline {
  color: lighten($col-a, 20);
  text-shadow: 1px 1px 2px rgba(black, 0.4);
  animation: tagline 1000ms ease-in forwards;

  @include bp(10) {
    font-size: font-size(m);
  }
}