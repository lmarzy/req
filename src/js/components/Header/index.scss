// ========================================================================
// JS/COMPONENTS/PARTIALS/HEADER
// ========================================================================

@import '~sassHelpers';

.header {
  display: flex;
  height: 90vh;
  align-items: center;
  justify-content: center;
  background-image: linear-gradient(to right top, $col-a, $col-b);
  clip-path: polygon(0 0, 100% 0, 100% 65%, 0 100%);
}

.logo {
  font-size: font-size(l);
  color: lighten($col-b, 35);
  text-shadow: 1px 1px 1px #000;

  @include bp(10) {
    font-size: font-size(xl);
  }
}