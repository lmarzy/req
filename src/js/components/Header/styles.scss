// ========================================================================
// JS/COMPONENTS/HEADER
// ========================================================================

@import '~sassHelpers';

.header {
  display: flex;
  height: 100vh;
  padding-top: 20vh;
  justify-content: center;
  background-image: linear-gradient(to right top, $col-a, $col-b);
  clip-path: polygon(0 0, 100% 0, 100% 80%, 0 100%);
}

.container {
  text-align: center;
}

.logo {
  font-size: 30px;
  color: white;
  font-weight: 700;
  text-transform: uppercase;

  @include bp(10) {
    font-size: 72px;
  }
}

.tagLine {
  font-size: 18px;
  color: rgba(white, 0.7);

  @include bp(10) {
    @include vr(24px, 2);
  }
}