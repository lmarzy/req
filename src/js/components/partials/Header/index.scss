// ========================================================================
// JS/COMPONENTS/PARTIALS/HEADER
// ========================================================================

@import '~sassHelpers';

.header {
  display: flex;
  padding-left: spacer(2);
  padding-right: spacer(2);
  border-bottom: 1px solid lightgray;
}

.header nav {
  margin-left: auto;
}

.logo {
  @include vr(20px, 2, 0);
}