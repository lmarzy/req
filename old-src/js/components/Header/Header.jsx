import React from 'react';
import PropTypes from 'prop-types';

import styles from './header.scss';

const Header = props => (
  <header className={styles.header}>
    <div className={styles.container}>
      <h1 className={styles.logo}>{props.siteName}</h1>
      <p className={styles.tagline}>{props.tagLine}</p>
    </div>
  </header>
);

Header.propTypes = {
  siteName: PropTypes.string.isRequired,
  tagLine: PropTypes.string.isRequired,
};

export default Header;
