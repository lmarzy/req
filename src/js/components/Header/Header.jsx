import React from 'react';
import PropTypes from 'prop-types';

import styles from './styles.scss';

const Header = ({ siteName, tagLine }) => (
  <header className={styles.header}>
    <div className={styles.container}>
      <h1 className={styles.logo}>{siteName}</h1>
      <p className={styles.tagLine}>{tagLine}</p>
    </div>
  </header>
);

Header.propTypes = {
  siteName: PropTypes.string.isRequired,
  tagLine: PropTypes.string.isRequired,
};

export default Header;
