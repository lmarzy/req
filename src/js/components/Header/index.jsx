import React from 'react';
import PropTypes from 'prop-types';

import styles from './index.scss';

const Header = props => (
  <header className={styles.header}>
    <h1 className={styles.logo}>{props.siteName}</h1>
  </header>
);

Header.propTypes = {
  siteName: PropTypes.string.isRequired,
};

export default Header;
