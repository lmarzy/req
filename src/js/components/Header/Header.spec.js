import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Header from './Header';

const component = shallow(<Header siteName="name" tagLine="tagline" />);
const tree = renderer.create(<Header siteName="name" tagLine="tagline" />).toJSON();

describe('<Header />', () => {
  it('Should render 1 <Header /> component', () => {
    expect(component).toHaveLength(1);
  });

  it('Should render a snapshot', () => {
    expect(tree).toMatchSnapshot();
  });
});
