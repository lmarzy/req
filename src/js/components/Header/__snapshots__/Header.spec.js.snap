// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render a snapshot 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <h1
      className="logo"
    >
      React Kickstarter
    </h1>
  </div>
</header>
`;
